app/models/talentlms/course.rb

```diff
@@ -60,5 +60,13 @@ def self.remove_course(course_id)
     def self.collection_for_form
       Talentlms::Course.all.map { |c| [c.name_with_subdomain, c.id] }
     end
+
+    # Check local data for course completion by user
+    def completed_by?(user)
+      login = Talentlms::Login.find_by(config: config, user: user)
+      return unless login # Login does not exist, user has not completed training
+
+      Talentlms::CompletedTraining.all.detect { |ct| ct.login_id == login.id && ct.course_id == id }.present?
```

Comment:
This looks like it is bringing back all the CompletedTraining records to determine if any match the login and course.  Couldn't it just:
```suggestion
      Talentlms::CompletedTraining.where(login_id: login.id, course_id: id).exists?
```

Reply:
yes, that's much better, adding it in.